front/components/items/Appointment/helper.ts

```diff
@@ -28,3 +24,20 @@ export const getDateOfAppointments = ({ appointments }: props) => {
         }
     });
 };
+
+export const returnAppointmentByID = ({
```

Comment:
it still says to me that you are returning appointment from list by its ID. there is need to add informations that you are filtering by other fields also.

filterAppointments or filterAppointmentsBy(and there your filters) or findMatchingAppointment.
give me info what did you choose